knowledge-base/adr/0031-resilience-options.md

```diff
@@ -0,0 +1,257 @@
+# Resilience Options in the SAP Cloud SDK for JavaScript
+
+## Status
+
+proposed
+
+## Context
+
+### Overview on Request Types
+
+If a user uses the typed clients or the `executeHttpRequest()` method, the SDK creates a different requests.
+In order to discuss the different resilience options it is good to list these requests first.
+The term `target system` is used for the system defined in the destination.
+
+- XSUAA (BTP service) to fetch a service token
+- Destiantaion service (BTP service) to fetch the destination
+- CSRF token request to the target system for non read requests
+- actual request to the target system
+
+In the discussion we group the latter two requests to the target system with respect to resilience and the first two.
```

Comment:
This sentence is incomplete, but I think I know what is meant. It would be easier to grasp this if you just omit this sentence and do this:

- BTP service requests:
  - ...
- Target system requests:
  - ...